### DISCUSSION TITLE
Importing playlists

### DISCUSSION THREAD

**Discussion by ruggeriruggero** [FreeTubeApp/FreeTube#4910] [Q&A]
First of all thank you so much for the new playlists feature, but it seems that it's still not possible to directly import the .csv files that youtube generates when you export your data, as Freetube only seems to accept .db files.
Is there any workaround for this? Any way to convert the .csv files into .db?

**Comment by efb4f5ff-1298-471a-8973-3d47447115dc** [comment]
Importing Playlists from YT isnt supported yet. There is no official way to convert this.

**Reply by papyrazzi** [reply]
Hello, I work on Linux Mint V22, and Freetube v0.21.3 Beta - In the settings, it is possible to download your subscriptions (it works) and your playlists, but no .csv files are present in the explorer of file (while they are good there).
What is the solution ?

**Reply by efb4f5ff-1298-471a-8973-3d47447115dc** [reply]
Hi @papyrazzi im not sure i understand your question. Is it possible for you to rephrase this

**Comment by TurtleDino** [comment]
This is from a different developer, but you could try https://pypi.org/project/freetube-import/ in the meantime.

**Comment by AtomicRobotMan0101** [comment]
+10 points for @TurtleDino 

Its exactly correct. Install freetube-import via the terminal using the appropriate python pypl installers (different linuxen do this differently, so generically...)... `pip install freetube-import`

I use Mint, so it prohibits pip install (the PEP668 thing) so one must use `pipx install freetube-import`

It does have two dependencies though: `tqdm` and `normalizer`.  The last may already be on thy system (use `pip list` to see if its there)... if not, install them via `pip install tqdm normalizer` (or `pipx`)

Usage is dead easy.

My exports are:

```
$ ls
'Excellent videos.csv'  'plasticity videos.csv'   playlists.csv  'python videos.csv'  'Watch later videos.csv'
```

usage is `$ freetube-import 'Excellent videos.csv'`

```
Reading file Excellent videos.csv, the playlistfile has 10 entries
writing to file Excellent videos.db
100%|  ███████████████████████████| 10/10 [00:09<00:00,  1.00it/s]
Task failed successfully! Excellent videos.db written, with 8 entries
```

The just go into Freetube and import the resulting DB file. Done!

**Comment by Stoneware7102** [comment]
I've made a python script with a LLM for doing this

You simply dump your YT playlist with takeout, as .cvs
The script will convert them into a .db that you can upload with freetube using the option for that

It's works perfectly apparently, a few click and a few seconds latter your hundred playlist are in Freetube
